src/bcc_sensor.c

```diff
@@ -824,7 +825,14 @@ int on_security_file_open(struct pt_regs *ctx, struct file *file)
 	if (!file) {
 		goto out;
 	}
-	if (!(file->f_flags & (O_CREAT | O_TRUNC))) {
+
+	if ((file->f_flags & (O_CREAT | O_TRUNC))) {
+		data.type = EVENT_FILE_CREATE;
+	}
+	else if (!(file->f_flags & (O_RDWR | O_WRONLY))) {
```

Comment:
Is NOT (!) operator needed here?

Reply:
It is, this is only reporting only 'read only' reads of existing files to help cut down on the number of opens that can be reported.  It's largely mimicking the behavior in the kernel implementation of this: https://github.com/vmware/kernel-event-collector-module/blob/master/src/file-hooks.c#L684